pkg/audit/verifier.go

```diff
@@ -46,25 +50,34 @@ type defaultDownloader struct {
 	transport transport.Client
 	overlay   *overlay.Cache
 	reporter
+
+	minBytesPerSecond memory.Size
 }
 
 // newDefaultDownloader creates a defaultDownloader
-func newDefaultDownloader(log *zap.Logger, transport transport.Client, overlay *overlay.Cache, id *identity.FullIdentity) *defaultDownloader {
-	return &defaultDownloader{log: log, transport: transport, overlay: overlay}
+func newDefaultDownloader(log *zap.Logger, transport transport.Client, overlay *overlay.Cache, id *identity.FullIdentity, minBytesPerSecond memory.Size) *defaultDownloader {
+	return &defaultDownloader{log: log, transport: transport, overlay: overlay, minBytesPerSecond: minBytesPerSecond}
 }
 
 // NewVerifier creates a Verifier
-func NewVerifier(log *zap.Logger, transport transport.Client, overlay *overlay.Cache, id *identity.FullIdentity) *Verifier {
-	return &Verifier{downloader: newDefaultDownloader(log, transport, overlay, id)}
+func NewVerifier(log *zap.Logger, transport transport.Client, overlay *overlay.Cache, id *identity.FullIdentity, minBytesPerSecond memory.Size) *Verifier {
+	return &Verifier{downloader: newDefaultDownloader(log, transport, overlay, id, minBytesPerSecond)}
 }
 
 // getShare use piece store client to download shares from nodes
 func (d *defaultDownloader) getShare(ctx context.Context, limit *pb.AddressedOrderLimit, stripeIndex int64, shareSize int32, pieceNum int) (share Share, err error) {
 	defer mon.Task()(&ctx)(&err)
 
+	bandwidthMsgSize := shareSize
+
+	// determines number of seconds allotted for receiving data from a storage node
+	seconds := time.Second * time.Duration(bandwidthMsgSize/d.minBytesPerSecond.Int32())
```

Comment:
should we be doing integer division for `bandwidthMsgSize/d.minBytesPerSecond`? It might be more precise if we do floating point division

Reply:
The way to avoid precision errors is to ensure that multiplication happens before division.

```
time.Duration(int(time.Second) * bandwidthMsgSize/d.minBytesPerSecond.Int()))
```

Reply:
so just convert them both to floats then divide?

Reply:
okay cool